mysqld_exporter.go

```diff
@@ -174,14 +266,47 @@ func (e *Exporter) setMetrics(scrapes <-chan []string) {
 			continue
 		}
 
-		if _, ok := e.metrics[name]; !ok {
-			e.metrics[name] = prometheus.NewGauge(prometheus.GaugeOpts{
-				Namespace: namespace,
-				Name:      name,
-			})
-		}
+		command := comRegex.FindStringSubmatch(name)
+		connection_error := connectionErrorRegex.FindStringSubmatch(name)
+		handler := handlerRegex.FindStringSubmatch(name)
+		innodb_buffer_pool := innodbBufferPoolRegex.FindStringSubmatch(name)
+		innodb_data := innodbDataRegex.FindStringSubmatch(name)
+		innodb_page := innodbPageRegex.FindStringSubmatch(name)
+		innodb_row := innodbRowRegex.FindStringSubmatch(name)
+		performance_schema := performanceSchemaRegex.FindStringSubmatch(name)
+		ssl := sslRegex.FindStringSubmatch(name)
+		thread := threadRegex.FindStringSubmatch(name)
+
+		if len(command) > 0 {
```

Comment:
This could look nicer with a switch:

```go
switch {
case len(command) > 0:
  // ...
case len(connectionError) > 0:
  // ...
case len(handler) > 0
  // ...

// ...

default:
  // ...
}
```